project/src/main/webapp/event-details.js

```diff
@@ -0,0 +1,82 @@
+const eventHost = 'test@example.com'; // hard-coded event host value
+let isLoggedIn = false;
+let currentUser = null;
+
+window.onload = function onLoad() {
+  checkLoginStatus();
+  getVolunteeringOpportunities();
+};
+
+
+/**
+ * Check the current login status to update
+ * isLoggedIn and currentUser.
+ */
+async function checkLoginStatus() {
+  const response = await fetch('/login-status');
+  const loginStatus = await response.json();
+  isLoggedIn = loginStatus.isLoggedIn;
+  currentUser = loginStatus.userEmail;
+  return loginStatus;
+}
+
+/**
+ * Adds the volunteering opportunities to the event card.
+ */
+async function getVolunteeringOpportunities() {
+  const response = await fetch('/event-volunteering-data');
+  const opportunities = await response.json();
+  for (const key in opportunities) {
+    if (opportunities.hasOwnProperty(key)) {
```

Comment:
would this not always be true?

Reply:
This is required by Google javascript style.